libhif/hif-source.c

```diff
@@ -1723,54 +1786,70 @@ hif_source_download_package (HifSource *source,
 		directory_slash = g_build_filename (directory, "/", NULL);
 	}
 
+	/* the number of packages to download */
+	hif_state_set_number_steps (state, packages->len);
+
 	/* is a local repo, i.e. we just need to copy */
```

Comment:
Not related to this patch but why *do* we copy for local repos?  I explicitly avoid that in rpm-ostree FWIW.